spec/lib/reference.spec.js

```diff
@@ -0,0 +1,76 @@
+"use strict";
+
+describe("lib/reference", () => {
+    var ArtifactMock, ArtifactSearchMock, authToken, dateService, hostMock, instance, logger, path, Reference, refName, requestPromiseMock, URI;
+
+    ArtifactMock = require("../mock/artifact-mock")();
+    ArtifactSearchMock = require("../mock/artifact-search-mock")();
+    authToken = "abcd1234";
+    dateService = require("../../lib/date-service")();
+    hostMock = "exampleHost";
+    logger = require("../../lib/logger")();
+    path = "some/example/path";
+    refName = "exampleRefName";
+    requestPromiseMock = require("../mock/request-promise-mock")();
+    URI = require("urijs");
+    Reference = require("../../lib/reference")(ArtifactMock, ArtifactSearchMock, dateService, logger, hostMock, requestPromiseMock, URI);
+
+    beforeEach(() => {
+        instance = new Reference(refName, authToken);
+    });
+    describe(".initArtifact()", () => {
```

Comment:
This goes for basically every test in this test file, I believe it is very important to test all the things that those objects get instantiated with. In fact, I think you shouldn't mock artifact at all and just let it instantiate one.